corehq/apps/es/es_query.py

```diff
@@ -473,6 +475,53 @@ def scroll_ids(self):
         """Returns a generator of all matching ids"""
         return self.exclude_source().scroll()
 
+    def scroll_ids_to_disk_and_iter_docs(self):
+        """Returns a ``ScanResult`` for all matched documents.
+
+        Used for iterating docs for a very large query where consuming the docs
+        via ``self.scroll()`` may exceed the amount of time that the scroll
+        context can remain open. This is achieved by:
+
+        1. Fetching the IDs for all matched documents (via ``scroll_ids()``) and
+           caching them in a temporary file on disk, then
+        2. fetching the documents by (chunked blocks of) IDs streamed from the
+           temporary file.
+
+        Original design PR: https://github.com/dimagi/commcare-hq/pull/20282
+
+        Caveats:
+        - There is no guarantee that the returned ScanResult's ``count``
+          property will match the number of yielded docs.
+        - Documents that are present when ``scroll_ids()`` is called, but are
+          deleted prior to being fetched in full will be missing from the
+          results, and this scenario will *not* raise an exception.
+        - If Elastic document ID values are ever reused (i.e. new documents
+          are created with the same ID of a previously-deleted document) then
+          this method would become unsafe because it could yield documents that
+          were not matched by the query.
```

Comment:
Is there any reason to suspect this might happen?  These are UUIDs and operate on the assumption that they will never be re-used.